### PR TITLE
[FIX] MP4 (from itunes) files not being processed (no subs) (#1223)

### PR DESCRIPTION
**Pull request by growupboron**
**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.
- [x] **I have mentioned this change in the [changelog](https://github.com/CCExtractor/ccextractor/blob/master/docs/CHANGES.TXT).**

**My familiarity with the project is as follows (check one):**

- [ ] I have never used CCExtractor.
- [ ] I have used CCExtractor just a couple of times.
- [x] I absolutely love CCExtractor, but have not contributed previously.
- [ ] I am an active contributor to CCExtractor.

---

Fixed the issue by regressing the changes, as discussed on the issue page, 0.85 release was working fine, on further investigation, 0.86 release was also correct. From 0.87 release, this issue started to pop up, so regressed the code back to 0.86 release for the full functionality.

### CODE REVIEW

File: src/thirdparty/gpacmp4/mp4.c [hunk 1]
```diff
@@ -528,189 +527,255 @@ int processmp4 (struct lib_ccx_ctx *ctx, struct ccx_s_mp4Cfg *cfg, char *file)
...
 					do_NAL(enc_ctx, dec_ctx, (unsigned char *)seqcnf->data, seqcnf->size, &dec_sub);
+					//do_NAL (dec_ctx, (unsigned char *) seqcnf->data, seqcnf->size, &dec_sub);
 				}
 			}
-			if (process_avc_track(ctx, file, f, i + 1, &dec_sub) != 0) {
-				mprint("Error on process_avc_track()\n");
+
+			if(process_avc_track(ctx, file, f, i + 1, &dec_sub) != 0)
+			{
+				mprint("error\n");
```
Comment (cfsmp3) [inline src/thirdparty/gpacmp4/mp4.c]: I pretty much prefer errors message to be specific. Error on process_avc_track() is much better than "error" (same everywhere else).

File: src/thirdparty/gpacmp4/mp4.c [hunk 2]
```diff
@@ -528,189 +527,255 @@ int processmp4 (struct lib_ccx_ctx *ctx, struct ccx_s_mp4Cfg *cfg, char *file)
...
-			(unsigned char)((type >> 8) % 0x100), (unsigned char)(type % 0x100),
-			(unsigned char)(subtype >> 24 % 0x100), (unsigned char)((subtype >> 16) % 0x100),
-			(unsigned char)((subtype >> 8) % 0x100), (unsigned char)(subtype % 0x100));
-
-		const u64 track_type = MEDIA_TYPE(type, subtype);
 
-		switch (track_type) {
-		case MEDIA_TYPE(GF_ISOM_MEDIA_VISUAL, GF_ISOM_SUBTYPE_XDVB): //vide:xdvb
+		if ( type == GF_ISOM_MEDIA_VISUAL && subtype == GF_ISOM_SUBTYPE_XDVB)
+		{
```
Comment (cfsmp3) [inline src/thirdparty/gpacmp4/mp4.c]: Is there a problem with MEDIA_TYPE? I think the switch() version is cleaner and having if's.

**Comment by NilsIrl** [comment]
@growupboron could you link the commit you're reverting?

**Comment by growupboron** [comment]
> IMHO going back to 0.86 is going to fix that regression but it's also going to remove some fixes that the original PR introduced, so I'd say it's a better idea to fix the actual bug that was introduced (which visually I can't see to be honest, so it will require some actual debugging).

Yes sir, I get the point.  I will get back to fixing this issue, once my mid terms get over in a week, sorry for the late reply. The actually debugging might take some time, but i look forward to the challenge.

File: src/thirdparty/gpacmp4/mp4.c (hunk 1 above)
Comment (growupboron) [reply src/thirdparty/gpacmp4/mp4.c]: Will keep in mind.

File: src/thirdparty/gpacmp4/mp4.c (hunk 2 above)
Comment (growupboron) [reply src/thirdparty/gpacmp4/mp4.c]: I think so, the problem might be with MEDIA_TYPE, I'll take of the switch() version, here.

**Comment by growupboron** [comment]
> @growupboron could you link the commit you're reverting?



> @growupboron could you link the commit you're reverting?

@NilsIrl I just cloned the 0.86 and regressed the code.

**Comment by cfsmp3** [comment]
Closing - seems to be abandoned?

**Pull request closed without merging**